pkg/storage/s3/s3.go

```diff
@@ -643,3 +718,58 @@ func (d *driver) RemoveStorage(cr *imageregistryv1.Config) (bool, error) {
 func (d *driver) ID() string {
 	return d.Config.Bucket
 }
+
+// sharedCredentialsFileFromSecret returns a location (path) to the shared credentials file that was created
+// using the provided secret.
+// Consumers can configure the aws Session using that file to use the credentials like:
+//
+// sharedCredentialsFile, err := sharedCredentialsFileFromSecret(secret)
+// if err != nil {
+// 	// handle error
+// }
+// options := session.Options{
+// 	SharedConfigState: session.SharedConfigEnable,
+// 	SharedConfigFiles: []string{sharedCredentialsFile},
+// }
+// sess := session.Must(session.NewSessionWithOptions(options))
+//
+// Note: the caller is responsible for cleaning up the created creds file.
+func sharedCredentialsFileFromSecret(secret *corev1.Secret) (string, error) {
+	switch {
+	case len(secret.Data["credentials"]) > 0:
+		return saveSharedCredentialsFile(secret.Data["credentials"])
+	case len(secret.Data["aws_access_key_id"]) > 0 && len(secret.Data["aws_secret_access_key"]) > 0:
+		accessKey := string(secret.Data["aws_access_key_id"])
+		secretKey := string(secret.Data["aws_secret_access_key"])
+		return sharedCredentialsFileFromStaticCreds(accessKey, secretKey)
+	default:
+		return "", fmt.Errorf("invalid secret for aws credentials")
+	}
+}
+
+func saveSharedCredentialsFile(data []byte) (string, error) {
+	f, err := ioutil.TempFile("", "aws-shared-credentials")
+	if err != nil {
+		return "", fmt.Errorf("failed to create file for shared credentials: %v", err)
+	}
+	defer f.Close()
+
+	if _, err := f.Write(data); err != nil {
+		return "", fmt.Errorf("failed to write credentials to %s: %v", f.Name(), err)
+	}
+	return f.Name(), nil
+}
+
+// sharedCredentialsFileFromStaticCreds takes an access/secret key pair and creates an AWS
+// configuration file using the provided keys.
+// It returns the path to the file containing the config data.
+func sharedCredentialsFileFromStaticCreds(accessKey string, accessSecret string) (string, error) {
+	buf := &bytes.Buffer{}
+	fmt.Fprint(buf, "[default]\n")
+	fmt.Fprintf(buf, "aws_access_key_id = %s\n", accessKey)
+	fmt.Fprintf(buf, "aws_secret_access_key = %s\n", accessSecret)
```

Comment:
Can it contain special symbols? Does the value need to be escaped?

Reply:
I can't find a definitive list of allowed characters for the secret key. I do see the occasional '+', '/', '[', ']' and parenthesis, but nothing that would require escaping. I'll update it to use a single Fprintf() with backticks just to be on the safe side.